test/functional/apps/discover/_doc_table.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const browser = getService('browser');
+  const log = getService('log');
+  const retry = getService('retry');
+  const esArchiver = getService('esArchiver');
+  const kibanaServer = getService('kibanaServer');
+  const queryBar = getService('queryBar');
+  const PageObjects = getPageObjects(['common', 'discover', 'header', 'timePicker']);
+  const defaultSettings = {
+    defaultIndex: 'logstash-*',
+  };
+
+  describe('discover doc table', function describeIndexTests() {
+    const defaultRowsLimit = 50;
+    const rowsHardLimit = 500;
+
+    before(async function () {
+      log.debug('load kibana index with default index pattern');
+      await esArchiver.load('discover');
+
+      // and load a set of makelogs data
+      await esArchiver.loadIfNeeded('logstash_functional');
+      await kibanaServer.uiSettings.replace(defaultSettings);
+      log.debug('discover doc table');
+      await PageObjects.common.navigateToApp('discover');
+    });
+
+    beforeEach(async function () {
+      await PageObjects.timePicker.setDefaultAbsoluteRange();
```

Comment:
You could set the default time range via UI setting before navigating to discover, it would save you a few seconds, and I think you're just changing it once, so you could reset it in the single test
https://github.com/elastic/kibana/blob/3250816d022f547b8fdb33dfcaa44df94846c80a/test/functional/page_objects/time_picker.ts#L65

Reply:
I used to do that but then a new test was introduced (`should refresh the table content when changing time window`) where it's changing the time window and it has a waterfall effect on other tests. :(

Reply:
and what if you execute `setDefaultAbsoluteRange` after work is done in this test? however, yes, it's cleaner to clean up after every step :) but when you develop those functional once for a while you're grateful for each second it runs faster 😄